icu4j/main/classes/core/src/com/ibm/icu/impl/number/MicroProps.java

```diff
@@ -17,16 +26,37 @@
     public DecimalSeparatorDisplay decimal;
     public IntegerWidth integerWidth;
 
-    // Populated by notation/unit:
+    // Modifiers provided by the number formatting pipeline (when the value is known):
+
+    // A Modifier provided by LongNameHandler, used for currency long names and
+    // units. If there is no LongNameHandler needed, this should be an
+    // null. (This is typically the third modifier applied.)
     public Modifier modOuter;
+
+    // A Modifier for short currencies and compact notation. (This is typically
+    // the second modifier applied.)
     public Modifier modMiddle;
+
+    // A Modifier provided by ScientificHandler, used for scientific notation.
+    // This is typically the first modifier applied.
     public Modifier modInner;
+
     public Precision rounder;
     public Grouper grouping;
     public boolean useCurrency;
 
     // Internal fields:
     private final boolean immutable;
+
+    // The MeasureUnit with which the output is represented. May also have
+    // MeasureUnit.Complexity.MIXED complexity, in which case mixedMeasures comes into
+    // play.
+    public MeasureUnit outputUnit;
+
+    // In the case of mixed units, this is the set of integer-only units
+    // *preceding* the final unit.
+    public List<Measure> mixedMeasures ;
```

Comment:
In C++, mixedMeasures only keeps a set of integers.

In this Java code, it's keeping the whole Measure - probably fine: we don't need it, but we have it, so might as well keep it? The number is extracted in MixedUnitLongNameHandler via `micros.mixedMeasures.get(i).getNumber()` - might as well extract the number there, rather than extract it at the time of populating mixedMeasures.

I think the difference may also be easy enough to figure out that it's okay if I don't document the difference in more detail.